stock_picking_auto_create_lot/models/stock_picking.py

```diff
@@ -11,16 +12,25 @@ def _set_auto_lot(self):
         """
         Allows to be called either by button or through code
         """
-        pickings = self.filtered(lambda p: p.picking_type_id.auto_create_lot)
-        lines = pickings.mapped("move_line_ids").filtered(
-            lambda x: (
-                not x.lot_id
-                and not x.lot_name
-                and x.product_id.tracking != "none"
-                and x.product_id.auto_create_lot
-            )
-        )
-        lines.set_lot_auto()
+        lines = self.env["stock.move.line"]
+        for picking in self:
+            if not picking.picking_type_id.auto_create_lot:
+                continue
+            domain = [
```

Comment:
I would have externalized the domain preparation in a hook method.

Then, having done the search outside the loop for performances reasons